File: count_substrings.c
#include <stdio.h>
#include <string.h>

int count(char s[], int start, int k) {
    if (start >= strlen(s))
        return 0;
    int freq[256] = {0}, unique = 0, count = 0;
    for (int end = start; end < strlen(s); end++) {
        if (freq[(int)s[end]] == 0)
            unique++;
        freq[(int)s[end]]++;
        if (unique == k)
            count++;
        else if (unique > k)
            break;
    }
    return count + count(s, start + 1, k);
}

int main() {
    char s[100];
    int k;
    printf("Enter string: ");
    scanf("%s", s);
    printf("Enter k: ");
    scanf("%d", &k);
    printf("%d\n", count(s, 0, k));
    return 0;
}

File: count_substrings.py
def count_substrings(s, k):
    def helper(start, unique, freq):
        if start == len(s):
            return 0
        count = 0
        for end in range(start, len(s)):
            ch = s[end]
            if freq.get(ch, 0) == 0:
                unique += 1
            freq[ch] = freq.get(ch, 0) + 1
            if unique == k:
                count += 1
            elif unique > k:
                break
        return count + helper(start + 1, 0, {})

    return helper(0, 0, {})

s = input("Enter string: ")
k = int(input("Enter k: "))
print(count_substrings(s, k))
